compiler/rustc_hir_analysis/src/hir_ty_lowering/dyn_trait.rs

```diff
@@ -80,8 +80,13 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
             span,
         );
 
-        let (elaborated_trait_bounds, elaborated_projection_bounds) =
+        let (mut elaborated_trait_bounds, elaborated_projection_bounds) =
             traits::expand_trait_aliases(tcx, user_written_bounds.iter().copied());
+
+        // FIXME(sized-hierarchy): https://github.com/rust-lang/rust/pull/142712#issuecomment-3013231794
+        let meta_sized_did = tcx.require_lang_item(LangItem::MetaSized, span);
+        elaborated_trait_bounds.retain(|(pred, _)| pred.def_id() != meta_sized_did);
```

Comment:
does this not also support explicit `dyn Clone + MetaSized` by stripping the metasized bound?

I feel like removing all `MetaSized` bounds might be generally correct here, unsure 🤷